scipy/linalg/tests/test_lapack.py

```diff
@@ -1705,3 +1705,267 @@ def test_getc2_gesc2():
+
+    gtsvx_out = gtsvx(dl, d, du, b, fact=fact, trans=trans, dlf=dlf_, df=df_,
+                      duf=duf_, du2=du2f_, ipiv=ipiv_)
+    dlf, df, duf, du2f, ipiv, x_soln, rcond, ferr, berr, info = gtsvx_out
+    # test with singular matrix
+    # no need to test with fact "F" since ?gttrf already tests for these.
+    if fact == "N":
+
+        L, U = gttrf_to_lu(n, dtype, dlf, df, duf, du2f, ipiv)
+        # check that the outputs define an LU decomposition of A
+        assert_allclose(A, L @ U, atol=atol, rtol=rtol)
+        d[-1] = 0
+        dl[-1] = 0
+
+        # solve using routine
+        gtsvx_out = gtsvx(dl, d, du, b)
+        dlf, df, duf, du2f, ipiv, x_soln, rcond, ferr, berr, info = gtsvx_out
+        # test for the singular matrix.
+        assert info > 0, "info should be > 0 for singular matrix"
+
+    elif fact == 'F':
+        # assuming that a singular factorization is input
+        df_[-1] = 0
+        duf_[-1] = 0
+        du2f_[-1] = 0
+
+        gtsvx_out = gtsvx(dl, d, du, b, fact=fact, dlf=dlf_, df=df_, duf=duf_,
+                          du2=du2f_, ipiv=ipiv_)
+        dlf, df, duf, du2f, ipiv, x_soln, rcond, ferr, berr, info = gtsvx_out
+        # info should not be zero and should provide index of illegal value
+        assert 0 < info <= n, "incorrect info for singular matrix"
+        assert info > 0, "info should be > 0 for singular matrix"
+
+
+@pytest.mark.parametrize("dtype,trans", list(zip(DTYPES, ["N"]*len(DTYPES)))
+                         + list(zip(REAL_DTYPES, ["T"]*len(REAL_DTYPES)))
+                         + list(zip(COMPLEX_DTYPES,
+                                    ["C"]*len(COMPLEX_DTYPES))))
+@pytest.mark.parametrize("fact,dl_d_du_lambda",
+                         [("F",
+                           lambda dl,
+                           d,
+                           du: get_lapack_funcs('gttrf', dtype=d.dtype)(dl,
+                                                                        d,
+                                                                        du)),
+                          ("N", lambda dl, d, du: (None, None, None, None,
+                                                   None, None))])
+def test_gtsvx_error_incompatible_size(dtype, trans, fact, dl_d_du_lambda):
+    np.random.seed(42)
+    # obtain routine
+    gtsvx = get_lapack_funcs('gtsvx', dtype=dtype)
+    # Generate random tridiagonal matrix A
+    n = 10
+    dl = generate_random_dtype_array((n-1,), dtype=dtype)
+    d = generate_random_dtype_array((n,), dtype=dtype)
+    du = generate_random_dtype_array((n-1,), dtype=dtype)
+    A = np.diag(dl, -1) + np.diag(d) + np.diag(du, 1)
+    # generate random solution x
+    x = generate_random_dtype_array((n, 2), dtype=dtype)
+    # create b from x for equation Ax=b
+    if trans == 'N':
+        b = A @ x
+    else:
+        b = (A.conj().T) @ x
+    # determine dlf, df, duf, du2f, and IPIV from lambda expression
```

Comment:
This comment tells me *what* the line below is doing. I can already see that from the line below. What I really need to understand from the comment is *why* are we doing that?

Reply:
The lambda expression was my idea for how to parametrize the `fact` value, so that instead of branching based on` fact == N or T` every time, we could use the lambda to get `dlf_, df_, duf_, du2f_, ipiv_, info_lambda` where when `fact = N`, the lambda returns Nones, and when `fact = T`, lambda returns the output of gttrf.

Reply:
@swallan can you respond to @Kai-Striega's suggestion by updating the comment?